frontend/src/app/features/work-packages/routing/wp-view-base/work-package-single-view.base.ts

```diff
@@ -67,6 +73,10 @@ export class WorkPackageSingleViewBase extends UntilDestroyedMixin {
 
   @InjectField() readonly hooks:HookService;
 
+  @InjectField() readonly actions$:ActionsService;
```

Comment:
Pardon my ignorance, why `@Injectfield`? Because it's a base class that gets extended?

Reply:
e.g. the `constructor` method of injecting is basically nice syntactic sugar but becomes cumbersome for super classes?

Reply:
Yes exactly. We can question the use of class inheritance for that component instead of a common service (or helper function), but if you use inheritance, constructor-based DI falls apart as you have to forward all parameters to super in the correct order. That's what @InjectField was made for.